src/app/BufferedReadCallback.h

```diff
@@ -56,7 +56,7 @@ class BufferedReadCallback : public ReadClient::Callback
      *  2. The path provided in aPath is similar to what is buffered but we've hit the end of the report.
      *
      */
-    CHIP_ERROR DispatchBufferedData(const ConcreteAttributePath & aPath, const StatusIB & aStatus, bool aEndOfReport = false);
+    CHIP_ERROR DispatchBufferedData(const ConcreteDataAttributePath & aPath, const StatusIB & aStatus, bool aEndOfReport = false);
```

Comment:
Again, why this change?  Doesn't this break things completely?

Specifically, say we get a "set to empty" followed by "append one item" for the same list attribute.  The `ConcreteDataAttributePath` is different; the `ConcreteAttributePath` is the same.

The old code, quite purposefully, would _not_ dispatch in this situation, since it has not gotten the whole list.  The new code will dispatch, as far as I can tell, giving the consumer an empty list (which is wrong).

Am I just missing something?

If we do want to try to make this change, it needs to be made in a separate PR.  And again, I don't understand how this is not breaking any tests...

Reply:
good catch, updated back to ConcreteAttributePath, thanks

Reply:
regarding CHIP_ERROR DispatchBufferedData(const ConcreteAttributePath & aPath... if using ConcreteAttributePath, it is still working with ConcreteDataAttributePath since ConcreteDataAttributePath inherit ConcreteAttributePath, but ConcreteDataAttributePath has not defined its == operator, and using the ==operator from AttributeDataPath, if (aPath == mBufferedPath) is still working there